zulip_bots/zulip_bots/bots/trivia_quiz/test_trivia_quiz.py

```diff
@@ -58,14 +60,55 @@ def test_answers(self) -> None:
         self.assertEqual(quiz['question'], 'Which class of animals are newts members of?')
         self.assertEqual(quiz['correct_letter'], 'A')
         self.assertEqual(quiz['answers']['D'], 'Mammals')
+        self.assertEqual(quiz['answered_options'], [])
+        self.assertEqual(quiz['ended'], False)
 
         # test incorrect answer
         with patch('zulip_bots.bots.trivia_quiz.trivia_quiz.get_quiz_from_id',
                    return_value=json.dumps(quiz)):
             self._test('answer Q001 B', '**WRONG!** B is not correct :disappointed:')
+            self._test('answer Q001 C', '**WRONG!** C is not correct :disappointed:')
 
         # test correct answer
         with patch('zulip_bots.bots.trivia_quiz.trivia_quiz.get_quiz_from_id',
                    return_value=json.dumps(quiz)):
             with patch('zulip_bots.bots.trivia_quiz.trivia_quiz.start_new_quiz') as mock_new_quiz:
                 self._test('answer Q001 A', '**CORRECT!** Amphibian :tada:')
+
+    def test_grade_questions(self) -> None:
+        bot_handler = StubBotHandler()
+        quiz_payload = read_bot_fixture_data('trivia_quiz', 'test_new_question')['response']
+        with patch('random.shuffle'):
+            quiz = get_quiz_from_payload(quiz_payload)
+
+        bot_handler.storage.put('Q001', json.dumps(quiz))
```

Comment:
For this kind of test, where the sequence of operations is pretty important, I think a long test is fine, so I'd lean toward Neil's first advice, which is to use comments to highlight the steps of the test.  Sometimes a a really good helper function inside the test can also help, so that the end of the test reads a little quicker.